### PR TITLE
Added test_lint.R to test code quality

### PR DESCRIPTION
**Pull request by terrytangyuan**
Just wanna upload this to repo first and we'll deal with all the lint issues later. @hetong007  Thanks.

### CODE REVIEW

File: R-package/tests/testthat/test_lint.R [hunk 1]
```diff
@@ -0,0 +1,27 @@
+context("Code is of high quality and lint free")
+test_that("Code Lint", {
+  skip_on_cran()
+  skip_on_travis()
```
Comment (hetong007) [inline R-package/tests/testthat/test_lint.R]: Maybe we can have it run in Travis check as well?

**Comment by terrytangyuan** [comment]
Yeah that's exactly the plan but we'll wait until all/most of the issues from lint analysis are fixed.

**Comment by hetong007** [comment]
So may I know what are the current issues now? Maybe you can open an issue to track it.

**Comment by terrytangyuan** [comment]
Okay I'll work on this soon. I'll ping you when I am done. It's mostly checking the code style, format, consistency, and quality of code.

**Comment by hetong007** [comment]
Merged, thanks

**Pull request merged**